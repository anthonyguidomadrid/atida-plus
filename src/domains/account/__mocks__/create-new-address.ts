export const createNewAddress = {
  redirect: 'some-path',
  salutation: 'string',
  firstName: 'string',
  lastName: 'string',
  address1: 'string',
  address3: 'string',
  zipCode: 'string',
  city: 'string',
  country: 'string',
  iso2Code: 'string',
  phone: 'string',
  isDefaultShipping: true,
  isDefaultBilling: true
}

export const createNewAddressPayload = {
  data: {
    type: 'customers',
    attributes: {
      salutation: 'string',
      firstName: 'string',
      lastName: 'string',
      address1: 'string',
      address3: 'string',
      zipCode: 'string',
      city: 'string',
      country: 'string',
      iso2Code: 'string',
      phone: 'string',
      isDefaultShipping: true,
      isDefaultBilling: true
    }
  }
} as const

export const UpdatedJWTAndRefreshTokens = {
  refreshToken: 'newRefreshToken',
  JWT: 'newJWT',
  expiresIn: '12345'
}

export const sprykerUpdatedJWTAndRefreshTokens = {
  data: {
    type: '',
    id: '',
    attributes: {
      tokenType: 'Bearer',
      expiresIn: '12345',
      accessToken: 'newJWT',
      refreshToken: 'newRefreshToken',
      idCompanyUser: ''
    },
    links: {
      self: ''
    }
  },
  links: {
    self: ''
  }
}
