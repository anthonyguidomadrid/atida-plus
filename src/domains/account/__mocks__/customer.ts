export const createCustomer = {
  salutation: 'Mr',
  firstName: 'Test',
  lastName: 'McTest',
  email: 'some@email.com',
  password: 'somePassword1!',
  confirmPassword: 'somePassword1!',
  acceptedTerms: true,
  emailNotification: true,
  phone: '654654654',
  company: null,
  surcharge: false,
  addresses: {
    addresses: [
      {
        salutation: 'Mr',
        firstName: 'Test',
        lastName: 'McTest',
        address1: 'Brugstraat 123',
        zipCode: '1011 AD',
        city: 'Amsterdam',
        iso2Code: 'PT',
        phone: '088273791298432',
        isDefaultShipping: true,
        isDefaultBilling: true
      }
    ]
  }
}

export const sprykerCustomer = {
  data: {
    type: 'customers',
    id: 'some-id',
    attributes: {
      salutation: 'Mr',
      firstName: 'Test',
      lastName: 'McTest',
      email: 'some@email.com',
      password: 'somePassword1!',
      confirmPassword: 'somePassword1!',
      acceptedTerms: true,
      emailNotification: true,
      company: null,
      surcharge: false
    }
  }
} as const

export const sprykerCustomerWithAddresses = {
  data: {
    ...sprykerCustomer,
    attributes: {
      ...sprykerCustomer.data.attributes
    },
    relationships: {
      addresses: {
        data: [
          { type: 'addresses', id: 'address-id-1' },
          { type: 'addresses', id: 'address-id-2' }
        ]
      }
    }
  },
  included: [
    {
      type: 'addresses',
      id: 'address-id-1',
      attributes: {
        salutation: 'some-salutation',
        firstName: 'some-name',
        lastName: 'some-last-name',
        address1: 'some-address',
        address2: 'some-address2',
        address3: 'some-address3',
        zipCode: 'some-zip-code',
        city: 'some-city',
        country: 'some-country',
        iso2Code: 'some-iso-code',
        company: 'some-company',
        phone: 'some-phone',
        isDefaultShipping: true,
        isDefaultBilling: true,
        isTaxExempt: false
      }
    },
    {
      type: 'addresses',
      id: 'address-id-2',
      attributes: {
        salutation: 'some-other-salutation',
        firstName: 'some-other-name',
        lastName: 'some-other-last-name',
        address1: 'some-other-address',
        address2: 'some-other-address2',
        address3: 'some-other-address3',
        zipCode: 'some-other-zip-code',
        city: 'some-other-city',
        country: 'some-other-country',
        iso2Code: 'some-other-iso-code',
        company: 'some-other-company',
        phone: 'some-other-phone',
        isDefaultShipping: false,
        isDefaultBilling: false,
        isTaxExempt: false
      }
    }
  ]
}
