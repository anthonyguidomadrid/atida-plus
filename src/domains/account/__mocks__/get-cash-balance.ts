export const balance = {
  total: 20
}
