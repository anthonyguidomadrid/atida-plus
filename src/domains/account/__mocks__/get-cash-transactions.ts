import { AtidaCashTransactionHistoryLedgerResponse } from '../types'

export const mockedTransactionHistory = {
  data: [
    {
      transaction_type: 'debit',
      amount: 200,
      transaction_date: '2022-10-04T14:20:22.029764445Z'
    },
    {
      transaction_type: 'credit',
      amount: 280,
      expiry_date: '2023-10-04T14:20:22.029764445Z',
      transaction_date: '2022-10-04T14:20:22.029764445Z'
    },
    {
      transaction_type: 'credit',
      amount: 280,
      expiry_date: '2023-07-23T14:20:22.029764445Z',
      transaction_date: '2022-07-23T14:20:22.029764445Z'
    },
    {
      transaction_type: 'debit',
      amount: 200,
      transaction_date: '2022-06-23T14:20:22.029764445Z'
    },
    {
      transaction_type: 'credit',
      amount: 280,
      expiry_date: '2023-06-23T14:20:22.029764445Z',
      transaction_date: '2022-06-23T14:20:22.029764445Z'
    },
    {
      transaction_type: 'credit',
      amount: 500,
      expiry_date: '2023-05-23T14:20:22.029764445Z',
      transaction_date: '2022-05-23T14:20:22.029764445Z'
    },
    {
      transaction_type: 'debit',
      amount: 280,
      expiry_date: '2023-04-23T14:20:22.029764445Z',
      transaction_date: '2022-04-23T14:20:22.029764445Z'
    },
    {
      transaction_type: 'credit',
      amount: 200,
      expiry_date: '2023-04-23T14:20:22.029764445Z',
      transaction_date: '2022-04-23T14:20:22.029764445Z'
    }
  ],
  metadata: {
    count: 8,
    total: 27
  }
} as AtidaCashTransactionHistoryLedgerResponse

export const mockedTransactionHistoryBayDateArray = {
  transactions: [
    {
      date: '2022-09-30',
      transactions: [
        {
          transaction_date: '2022-09-30T13:36:27.297Z',
          transaction_type: 'debit',
          amount: 200
        },
        {
          amount: 280,
          expiry_date: '2023-09-30T13:36:27.297Z',
          transaction_date: '2022-09-30T13:36:27.297Z',
          transaction_type: 'credit'
        },
        {
          transaction_type: 'debit',
          amount: 200,
          transaction_date: '2022-09-30T13:36:27.297Z'
        },
        {
          transaction_date: '2022-09-30T13:36:27.297Z',
          transaction_type: 'credit',
          amount: 180,
          expiry_date: '2023-09-30T13:36:27.297Z'
        },
        {
          amount: 100,
          transaction_date: '2022-09-30T13:36:27.297Z',
          transaction_type: 'debit'
        },
        {
          transaction_date: '2022-09-30T13:36:27.297Z',
          transaction_type: 'credit',
          amount: 670,
          expiry_date: '2023-09-30T13:36:27.297Z'
        }
      ]
    },
    {
      date: '2022-07-23',
      transactions: [
        {
          transaction_date: '2022-07-23T14:20:22.029764445Z',
          transaction_type: 'debit',
          amount: 200
        },
        {
          expiry_date: '2023-07-23T14:20:22.029764445Z',
          transaction_date: '2022-07-23T14:20:22.029764445Z',
          transaction_type: 'credit',
          amount: 280
        },
        {
          transaction_date: '2022-07-23T14:20:22.029764445Z',
          transaction_type: 'debit',
          amount: 200
        },
        {
          amount: 290,
          expiry_date: '2023-07-23T14:20:22.029764445Z',
          transaction_date: '2022-07-23T14:20:22.029764445Z',
          transaction_type: 'credit'
        }
      ]
    },
    {
      date: '2022-06-23',
      transactions: [
        {
          transaction_type: 'debit',
          amount: 200,
          transaction_date: '2022-06-23T14:20:22.029764445Z'
        },
        {
          expiry_date: '2023-06-23T14:20:22.029764445Z',
          transaction_date: '2022-06-23T14:20:22.029764445Z',
          transaction_type: 'credit',
          amount: 280
        }
      ]
    },
    {
      date: '2022-05-23',
      transactions: [
        {
          transaction_type: 'credit',
          amount: 500,
          expiry_date: '2023-05-23T14:20:22.029764445Z',
          transaction_date: '2022-05-23T14:20:22.029764445Z'
        }
      ]
    },
    {
      date: '2022-04-23',
      transactions: [
        {
          transaction_date: '2022-04-23T14:20:22.029764445Z',
          transaction_type: 'debit',
          amount: 280,
          expiry_date: '2023-04-23T14:20:22.029764445Z'
        },
        {
          transaction_type: 'credit',
          amount: 200,
          expiry_date: '2023-04-23T14:20:22.029764445Z',
          transaction_date: '2022-04-23T14:20:22.029764445Z'
        }
      ]
    }
  ],
  totalPages: 4
}
