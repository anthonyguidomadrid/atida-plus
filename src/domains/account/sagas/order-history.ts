import { SagaIterator } from '@redux-saga/types'
import type { AxiosResponse } from 'axios'
import { call, getContext, put, takeLatest } from 'redux-saga/effects'
import { createClient, logger } from '~helpers'
import {
  orderHistoryFailure,
  orderHistoryFulfill,
  orderHistoryRequest,
  orderHistorySuccess,
  orderHistoryTrigger
} from '../slices'
import type {
  OrderHistoryPayload,
  OrderHistoryTriggerPayload
} from '~domains/account/types'
import type { PayloadAction } from '@reduxjs/toolkit'
import { getErrorMessage } from '~helpers/error'

function* doOrderHistorySaga({
  payload
}: PayloadAction<OrderHistoryTriggerPayload>) {
  try {
    yield put(orderHistoryRequest())

    const locale: string = yield getContext('locale')
    const client = createClient({ locale, addAnonymousCustomerUniqueId: false })

    const apiResponse: AxiosResponse<OrderHistoryPayload> = yield call(
      client.get,
      '/api/account/order-history',
      { params: payload }
    )

    yield put(orderHistorySuccess(apiResponse.data))

    return apiResponse
  } catch (error) {
    const errorMessages = getErrorMessage(error)
    logger.error(errorMessages)
    if (Array.isArray(errorMessages)) {
      yield put(orderHistoryFailure({ message: errorMessages[0] }))
    } else {
      yield put(
        orderHistoryFailure({
          message: errorMessages
        })
      )
    }
  } finally {
    yield put(orderHistoryFulfill())
  }
}

export function* orderHistorySaga(): SagaIterator {
  yield takeLatest(orderHistoryTrigger, doOrderHistorySaga)
}
