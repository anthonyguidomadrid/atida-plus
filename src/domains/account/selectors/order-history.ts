import { createSelector } from '@reduxjs/toolkit'
import { ORDER_FAILED_STATUSES } from '~config/constants/order-status'
import { RootState } from '~domains/redux'

const selectOrderHistory = (state: RootState) =>
  state?.client?.account?.['order-history']

export const selectIsLoading = createSelector(
  selectOrderHistory,
  orderHistory => orderHistory?.isLoading ?? false
)

export const selectWasSuccess = createSelector(
  selectOrderHistory,
  orderHistory => orderHistory?.wasSuccess ?? false
)

export const selectAllOrdersFromHistory = createSelector(
  selectOrderHistory,
  orderHistory => orderHistory?.orderHistory
)

export const selectAllNotCancelledOrdersFromHistory = createSelector(
  selectOrderHistory,
  orderHistory =>
    orderHistory?.orderHistory
      ?.map(orderHistoryByDate => {
        const badge = orderHistoryByDate.orderHistory.filter(
          order =>
            !ORDER_FAILED_STATUSES.includes(order.attributes.status[0].name)
        )
        if (badge.length > 0)
          return {
            date: orderHistoryByDate.date,
            unFormattedDate: orderHistoryByDate.unFormattedDate,
            orderHistory: badge
          }
      })
      .filter(orderHistoryByDate => orderHistoryByDate)
)

export const selectOrderHistoryTotalPages = createSelector(
  selectOrderHistory,
  orderHistory => orderHistory?.totalPages
)

export const selectNumberOfOrdersFromHistory = createSelector(
  selectOrderHistory,
  orderHistory =>
    Array.isArray(orderHistory?.orderHistory?.[0]?.orderHistory)
      ? orderHistory.orderHistory?.reduce((accumulator, { orderHistory }) => {
          return accumulator + (orderHistory?.length ?? 0)
        }, 0)
      : 0
)
