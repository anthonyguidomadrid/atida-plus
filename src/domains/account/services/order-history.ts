import getConfig from 'next/config'
import { createClient } from '~helpers'
import {
  createElasticSearchClient,
  getElasticSearchProductIndex
} from '~domains/elasticsearch/helpers/client'
import {
  ElasticSearchProduct,
  normalizeElasticSearchProduct
} from '~domains/product'
import {
  OrderHistoryParams,
  OrderHistoryResponse,
  OrderItem,
  OrdersPayload
} from '../types'
import type { DateTimeFormat } from '~helpers'
if (typeof window !== 'undefined') {
  throw new Error('This file should not be called client-side')
}
export const orderHistory = async ({
  locale,
  accessToken,
  params
}: OrderHistoryParams): Promise<OrderHistoryResponse> => {
  const { serverRuntimeConfig } = getConfig()
  const client = createClient({
    locale,
    options: { baseURL: serverRuntimeConfig.ecommerceHost[locale] }
  })
  const pageLimit = 5
  /**
    We substract 1 to calculate properly the pageOffset:
      page = 1
      pageOffset = (1 - 1) * 5 = 0   --->  get orders between positions 0 to 5
      page = 2
      pageOffset = (2 - 1) * 5 = 5   --->  get orders between positions 5 to 10
      page = 3
      pageOffset = (3 - 1) * 5 = 10  --->  get orders between positions 10 to 15
      ...
  **/
  const pageOffset = params?.page ? (params.page - 1) * pageLimit : 0
  const response = await client.get<OrdersPayload>('/orders', {
    headers: {
      Authorization: `Bearer ${accessToken}`
    },
    params: {
      'page[limit]': pageLimit,
      'page[offset]': pageOffset,
      sort: params?.sort ?? '-created_at'
    }
  })
  const { data, links } = response.data
  if (data?.length === 0 || !data) {
    return { data, links }
  }
  const ordersWithItems = data.filter(item => item.attributes.items.length > 0)
  const ordersWithoutItems = data.filter(
    item => item.attributes.items.length === 0
  )

  if (ordersWithItems.length > 0) {
    const elasticClient = createElasticSearchClient()
    const elasticResponse = await elasticClient.mget<{
      docs: ElasticSearchProduct[]
    }>({
      index: getElasticSearchProductIndex(locale),
      body: {
        ids: Array.from(
          new Set(
            ordersWithItems.reduce<string[]>(
              (memo, item) =>
                memo.concat(item.attributes.items.map(i => i.sku)),
              []
            )
          )
        )
      }
    })
    const elasticData = ordersWithItems
      .reduce<OrderItem[]>(
        (memo, { attributes: { items } }) => [...memo, ...items],
        []
      )
      .map(item => {
        const doc = elasticResponse?.body?.docs?.find(
          doc => doc._id === item.sku
        )
        const isPromo = item.isPromo
        item.id = isPromo ? `${item.sku}-promotion` : item.sku
        doc && (doc.id = item.id)
        return normalizeElasticSearchProduct(
          locale,
          doc,
          undefined,
          undefined,
          isPromo
        )
      })
    const history = [...ordersWithItems, ...ordersWithoutItems]
      .map(oh => {
        if (!elasticData) return oh
        const {
          attributes: { items, ...otherAttributes },
          ...properties
        } = oh
        const itemsWithMetadata = items.map(item => {
          const { id } = item
          const { sku: elasticDataItemSku, ...elasticDataItem } =
            elasticData.find(eld => eld.id === id) ?? {}
          if (elasticDataItem) {
            return {
              ...item,
              metadata: elasticDataItem
            }
          }
          return item
        })
        return {
          ...properties,
          attributes: {
            items: itemsWithMetadata,
            ...otherAttributes
          }
        }
      })
      .reduce((memo, item) => {
        const dateTimeFormat: DateTimeFormat = {
          dateStyle: 'long'
        }
        const date = Intl.DateTimeFormat(locale, dateTimeFormat).format(
          new Date(item.attributes.createdAt)
        )
        const memoItems = memo.find(
          // @ts-ignore TODO: this needs typing, I don't understand it well enough
          m => m.date === date && Array.isArray(m.orderHistory)
        )
        if (memoItems) {
          // @ts-ignore TODO: this needs typing, I don't understand it well enough
          memoItems.orderHistory.push(item)
        } else {
          memo.push({
            // @ts-ignore TODO: this needs typing, I don't understand it well enough
            date,
            // @ts-ignore TODO: this needs typing, I don't understand it well enough
            unFormattedDate: item.attributes.createdAt,
            // @ts-ignore TODO: this needs typing, I don't understand it well enough
            orderHistory: [item]
          })
        }
        return memo
      }, [])

    return { data: history, links }
  }

  return {
    data: ordersWithoutItems.reduce((memo, item) => {
      const dateTimeFormat: DateTimeFormat = {
        dateStyle: 'long'
      }
      const date = Intl.DateTimeFormat(locale, dateTimeFormat).format(
        new Date(item.attributes.createdAt)
      )
      const memoItems = memo.find(
        // @ts-ignore TODO: this needs typing, I don't understand it well enough
        m => m.date === date && Array.isArray(m.orderHistory)
      )
      if (memoItems) {
        // @ts-ignore TODO: this needs typing, I don't understand it well enough
        memoItems.orderHistory.push(item)
      } else {
        memo.push({
          // @ts-ignore TODO: this needs typing, I don't understand it well enough
          date,
          // @ts-ignore TODO: this needs typing, I don't understand it well enough
          unFormattedDate: item.attributes.createdAt,
          // @ts-ignore TODO: this needs typing, I don't understand it well enough
          orderHistory: [item]
        })
      }
      return memo
    }, []),
    links
  }
}
