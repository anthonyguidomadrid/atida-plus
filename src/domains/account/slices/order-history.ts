import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'
import type {
  OrderHistoryPayload,
  OrderHistorySortedDates,
  OrderHistoryTriggerPayload
} from '../types'

type OrderHistoryState = {
  isLoading: boolean
  wasSuccess: boolean
  wasError: boolean
  error?: string
  orderHistory?: OrderHistorySortedDates[]
  totalPages?: number
}

const initialState: OrderHistoryState = {
  isLoading: false,
  wasSuccess: false,
  wasError: false
}

export const orderHistorySlice = createSlice({
  name: 'account/order-history',
  initialState,
  reducers: {
    trigger(_state, _action: PayloadAction<OrderHistoryTriggerPayload>) {
      // no-op, used for triggering saga
    },
    request(state) {
      state.isLoading = true
      state.wasSuccess = false
      state.wasError = false

      delete state.error
    },
    success(state, action: PayloadAction<OrderHistoryPayload>) {
      const { data, totalPages } = action.payload

      state.wasSuccess = true
      state.orderHistory = data
      state.totalPages = totalPages
    },
    failure(state, action: PayloadAction<{ message: string }>) {
      const { message } = action.payload

      state.wasError = true
      state.error = message
    },
    fulfill(state) {
      state.isLoading = false
    },
    clearDetails(state) {
      delete state.orderHistory
      delete state.totalPages
    }
  }
})
