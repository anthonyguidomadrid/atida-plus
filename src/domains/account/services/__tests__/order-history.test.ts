/**
 * @jest-environment node
 */
import axios from 'axios'
// @ts-ignore
import { mget } from '@elastic/elasticsearch'
import { elasticsearchProducts } from '~domains/basket/__mocks__/basket'
import { createClient } from '~helpers'
import { orderHistory } from '../order-history'
import {
  orderHistoryOnlyPending,
  orderHistoryWithoutItems
} from '~domains/account/__mocks__/order-history'

describe(orderHistory, () => {
  describe('creates the client and passes the locale to the spryker request', () => {
    beforeEach(() => {
      ;(axios.get as jest.Mock).mockResolvedValueOnce({
        data: {
          data: [],
          links: []
        }
      })
    })

    it('without pagination data', async () => {
      await orderHistory({
        locale: 'en-gb',
        accessToken: 'some-token'
      })

      expect(createClient).toHaveBeenCalledWith({
        locale: 'en-gb',
        options: {
          baseURL: 'spryker.co.uk'
        }
      })

      expect(axios.get).toHaveBeenCalledWith('/orders', {
        headers: {
          Authorization: 'Bearer some-token'
        },
        params: {
          'page[limit]': 5,
          'page[offset]': 0,
          sort: '-created_at'
        }
      })
    })

    it('with pagination data', async () => {
      await orderHistory({
        locale: 'en-gb',
        accessToken: 'some-token',
        params: { page: 3 }
      })

      expect(createClient).toHaveBeenCalledWith({
        locale: 'en-gb',
        options: {
          baseURL: 'spryker.co.uk'
        }
      })

      expect(axios.get).toHaveBeenCalledWith('/orders', {
        headers: {
          Authorization: 'Bearer some-token'
        },
        params: {
          'page[limit]': 5,
          'page[offset]': 10,
          sort: '-created_at'
        }
      })
    })
  })
  describe('test order history', () => {
    it('with populated data', async () => {
      mget.mockResolvedValue(elasticsearchProducts)
      ;(axios.get as jest.Mock).mockResolvedValue({
        data: {
          data: orderHistoryOnlyPending[0].orderHistory,
          links: []
        }
      })
      const res = await orderHistory({
        locale: 'en-gb',
        accessToken: 'some-token',
        params: { page: 1 }
      })
      if (res.data) {
        expect(res).toMatchSnapshot()
      }
    })
    it('without order history items', async () => {
      mget.mockResolvedValue(elasticsearchProducts)
      ;(axios.get as jest.Mock).mockResolvedValue({
        data: {
          data: orderHistoryWithoutItems[0].orderHistory,
          links: []
        }
      })
      const res = await orderHistory({
        locale: 'en-gb',
        accessToken: 'some-token',
        params: { page: 1 }
      })
      if (res.data) {
        expect(res).toMatchSnapshot()
      }
    })
  })
})
