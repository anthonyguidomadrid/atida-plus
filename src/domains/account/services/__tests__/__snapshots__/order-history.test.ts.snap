// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderHistory test order history with populated data 1`] = `
Object {
  "data": Array [
    Object {
      "date": "1 October 2021",
      "orderHistory": Array [
        Object {
          "attributes": Object {
            "calculatedDiscounts": Array [],
            "createdAt": "2021-10-01 07:41:49.623969",
            "currencyIsoCode": "EUR",
            "items": Array [
              Object {
                "id": "846502351931",
                "isPromo": false,
                "metadata": Object {},
                "quantity": 1,
                "sku": "846502351931",
                "sumGrossPrice": 942,
                "unitGrossPrice": 942,
              },
            ],
            "priceMode": "GROSS_MODE",
            "productDiscounts": Array [],
            "shipments": Array [
              Object {
                "erpShipmentStatus": "in_progres",
                "erpTrackingReference": "some-tracking-url",
                "erpTrackingUrl": "some-tracking-url",
                "estimatedDeliveryDate": null,
                "maxDeliveryDays": 3,
                "minDeliveryDays": 1,
                "shipmentStatus": null,
                "trackingLink": null,
              },
            ],
            "status": Array [
              Object {
                "displayName": "oms.state.pending",
                "name": "pending",
              },
            ],
            "totals": Object {
              "canceledTotal": 0,
              "discountTotal": 0,
              "expenseTotal": 399,
              "grandTotal": 1341,
              "remunerationTotal": 0,
              "subtotal": 942,
              "surchargeTotal": 0,
              "taxTotal": 240,
            },
          },
          "id": "PT--8723",
          "links": Object {
            "self": "http://glue-es-spryker.dev.atida.com/orders/PT--8723?page[limit]=5&page[offset]=0&sort=-created_at",
          },
          "maxProductsIndex": 7,
          "type": "orders",
        },
      ],
      "unFormattedDate": "2021-10-01 07:41:49.623969",
    },
  ],
  "links": Array [],
}
`;

exports[`orderHistory test order history without order history items 1`] = `
Object {
  "data": Array [
    Object {
      "date": "24 February 2021",
      "orderHistory": Array [
        Object {
          "attributes": Object {
            "calculatedDiscounts": Array [],
            "createdAt": "2021-02-24T10:07:27+00:00",
            "currencyIsoCode": "EUR",
            "items": Array [],
            "priceMode": "GROSS_MODE",
            "productDiscounts": Array [],
            "shipments": Array [
              Object {
                "erpShipmentStatus": "in_progres",
                "erpTrackingReference": "some-tracking-url",
                "erpTrackingUrl": "some-tracking-url",
                "estimatedDeliveryDate": null,
                "maxDeliveryDays": 3,
                "minDeliveryDays": 1,
                "shipmentStatus": null,
                "trackingLink": null,
              },
            ],
            "status": Array [
              Object {
                "displayName": "oms.state.pending",
                "name": "pending",
              },
            ],
            "totals": Object {
              "canceledTotal": 0,
              "discountTotal": 0,
              "expenseTotal": 399,
              "grandTotal": 1341,
              "remunerationTotal": 0,
              "subtotal": 942,
              "surchargeTotal": 0,
              "taxTotal": 240,
            },
          },
          "id": "PT--8723",
          "links": Object {
            "self": "http://glue-es-spryker.dev.atida.com/orders/PT--8723?page[limit]=5&page[offset]=0&sort=-created_at",
          },
          "maxProductsIndex": 7,
          "type": "orders",
        },
      ],
      "unFormattedDate": "2021-02-24T10:07:27+00:00",
    },
  ],
  "links": Array [],
}
`;
